library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileOutputOperator.java

```diff
@@ -1143,8 +1149,12 @@ protected void finalizeFile(String fileName) throws IOException
     for (FileStatus status : statuses) {
       String statusName = status.getPath().getName();
       if (statusName.endsWith(TMP_EXTENSION) && statusName.startsWith(fileName)) {
-        LOG.debug("deleting vagrant file {}", statusName);
-        fs.delete(status.getPath(), true);
+        //a tmp file has tmp extension always preceded by timestamp
+        String actualFileName = statusName.substring(0, statusName.lastIndexOf('.', statusName.lastIndexOf('.') - 1));
```

Comment:
The probability of a random file which ends with .tmp and starts with an actual file name used by this is just too low. This method can be overridden though, it someone using it face such issues